src/library/dao/trackdao.cpp

```diff
@@ -745,13 +741,13 @@ QList<TrackId> TrackDAO::addTracks(const QList<QFileInfo>& fileInfoList,
         int addIndex = query.value(addIndexColumn).toInt();
         QString filePath = query.value(locationColumn).toString();
         const QFileInfo& fileInfo = fileInfoList.at(addIndex);
-        TrackInfoObject* pTrack = new TrackInfoObject(fileInfo);
-        addTracksAdd(pTrack, unremove);
+        TrackPointer pTrack(TrackInfoObject::newTemporaryForFile(fileInfo));
```

Comment:
It would even be possible to use a local variable allocated on the stack 
here if performance really matters. The shared pointer is only used locally 
and not passed around. Since TrackDAO is already a friend of TrackInfoObject 
this would be an option that also documents that the object is only used 
locally.

On the other hand I'm thinking about managing all file access through the 
cache, even for tracks that are not stored in the library. We want to be 
sure that each file access in Mixxx is managed to avoid race conditions when 
reading from or writing to audio files.

I need to do more testing before making a decision.

On 09/03/2015 10:37 AM, Daniel Schürmann wrote:
>
> In src/library/dao/trackdao.cpp 
> <https://github.com/mixxxdj/mixxx/pull/695#discussion_r38623757>:
>
> > @@ -745,13 +741,13 @@ QList<TrackId> TrackDAO::addTracks(const QList<QFileInfo>& fileInfoList,
> >          int addIndex = query.value(addIndexColumn).toInt();
> >          QString filePath = query.value(locationColumn).toString();
> >          const QFileInfo& fileInfo = fileInfoList.at(addIndex);
> > -        TrackInfoObject* pTrack = new TrackInfoObject(fileInfo);
> > -        addTracksAdd(pTrack, unremove);
> > +        TrackPointer pTrack(TrackInfoObject::newTemporaryForFile(fileInfo));
>
> After looking around a bit, I think it is correct to use 
> "newTemporaryForFile" here because of speed.
> Do you see a chance to loose meta-data changes in other temporary 
> instances? I don't. Since this is used for mass import.
>
> Please verify this and place a comment. Thank you.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/695/files#r38623757>.
>